core/iwasm/common/gc/gc_object.c

```diff
@@ -226,31 +225,43 @@ wasm_array_obj_new(WASMExecEnv *exec_env, WASMRttTypeRef rtt_type,
 
     array_obj->header = (WASMObjectHeader)rtt_type;
     array_obj->length = (length << 2) | elem_size_log;
-    for (i = 0; i < length; i++) {
-        if (wasm_is_type_reftype(array_type->elem_type)) {
-            uint32 *elem_addr =
-                (uint32 *)array_obj->elem_data + REF_CELL_NUM * i;
-            PUT_REF_TO_ADDR(elem_addr, init_value->gc_obj);
-        }
-        else if (array_type->elem_type == VALUE_TYPE_I32
-                 || array_type->elem_type == VALUE_TYPE_F32) {
-            ((int32 *)array_obj->elem_data)[i] = init_value->i32;
-        }
-        else if (array_type->elem_type == PACKED_TYPE_I8) {
-            ((int8 *)array_obj->elem_data)[i] = (int8)init_value->i32;
-        }
-        else if (array_type->elem_type == PACKED_TYPE_I16) {
-            ((int16 *)array_obj->elem_data)[i] = (int16)init_value->i32;
-        }
-        else {
-            uint32 *elem_addr = (uint32 *)array_obj->elem_data + 2 * i;
-            PUT_I64_TO_ADDR(elem_addr, init_value->i64);
+
+    if (init_value != NULL) {
```

Comment:
For array.new_fixed, the element values are provided through opcode, we don't need to init the array during creating.
So we use init_value = NULL to avoid filling the whole array